lib/csv/parser.rb

```diff
@@ -527,7 +527,7 @@ def prepare_separators
 
       @cr = "\r".encode(@encoding)
       @lf = "\n".encode(@encoding)
-      @cr_or_lf = Regexp.new("[\r\n]".encode(@encoding))
+      @cr_or_lf = Regexp.new("\r\n|\n|\r".encode(@encoding))
```

Comment:
`@cr_or_lf` isn't match with this change. Could you rename this variable?

```suggestion
      @line_end = Regexp.new("\r\n|\n|\r".encode(@encoding))
```